src/geo/projection/globe_util.js

```diff
@@ -724,37 +730,117 @@ export class GlobeSharedBuffers {
-            const latitudinalLod = GLOBE_LATITUDINAL_GRID_LOD_TABLE[k];
-            for (let j = 0; j < latitudinalLod; j++) {
-                for (let i = 0; i < quadExt; i++) {
-                    const index = j * vertexExt + i;
-                    gridIndices.emplaceBack(index + 1, index, index + vertexExt);
-                    gridIndices.emplaceBack(index + vertexExt, index + vertexExt + 1, index + 1);
+    // Generate terrain grid vertices and indices for all LOD's
+    //
+    // Grid vertices memory layout:
+    //
+    //          First line strip
+    //          ┌───────────────┐
+    //          │┌─────────────┐│
+    // Left     ││┼┼┼┼┼┼┼┼┼┼┼┼┼││ Right
+    // Border   ││┼┼┼┼┼┼┼┼┼┼┼┼┼││ Border
+    // Strip    │├─────────────┤│ Strip
+    //          ││  Main Grid  ││
+    //          │├─────────────┤│
+    //          ││┼┼┼┼┼┼┼┼┼┼┼┼┼││
+    //          ││┼┼┼┼┼┼┼┼┼┼┼┼┼││
+    //          │└─────────────┘│
+    //          ├───────────────┤
+    //          ├───────────────┤
+    //          └───────────────┘
+    //      Bottom strips = Number of LOD's
+    //
+    // export type GridWithLods = {
+    //     gridVertices: PosGlobeExtArray,
+    //     gridIndices: TriangleIndexArray,
+    //     gridSegments: Array<SegmentVector>
+    // };
+    _fillGridMeshWithLods(longitudinalCellsCount: number, latitudinalLods: number[], hasStrip: boolean): GridWithLods {
+        const vertices = new PosGlobeExtArray();
+        const indices = new TriangleIndexArray();
+        const segments: Array<SegmentVector> = [];
+
+        const xVertices = longitudinalCellsCount + 1 + 2 * (hasStrip ? 1 : 0);
+        const yVerticesHighLodNoStrip = latitudinalLods[0] + 1;
+        const yVerticesHighLodWithStrip = latitudinalLods[0] + 1 + (hasStrip ? 1 + latitudinalLods.length : 0);
+
+        // Index adjustment, used to make strip (x, y) vertex input attribute data
+        // to match same data on ordinary grid edges
+        const xCoordAdjusted = (idx: number) => {
+            if (!hasStrip) return idx;
+
+            if (idx === xVertices - 1) {
+                return idx - -2;
+            } else if (idx === 0) {
+                return idx;
+            } else {
+                return idx - 1;
+            }
+        };
+
+        // Add first horizontal strip if present
+        if (hasStrip) {
+            for (let x = 0; x < xVertices; ++x) {
+                vertices.emplaceBack(xCoordAdjusted(x), 0, 1);
+            }
+        }
+
+        // Add main grid part with vertices strips embedded
+        for (let y = 0; y < yVerticesHighLodNoStrip; ++y) {
+            for (let x = 0; x < xVertices; ++x) {
+                const isSideBorder = (x === 0 || x === xVertices - 1);
+                let stripFactor = 0;
+                if (isSideBorder && hasStrip) {
+                    stripFactor = 1;
                 }
+
+                vertices.emplaceBack(xCoordAdjusted(x), y, stripFactor);
```

Comment:
Nit: we could use some ternaries in a few places here to make the code simpler, e.g.:

```suggestion
                vertices.emplaceBack(xCoordAdjusted(x), y, isSideBorder && hasStrip ? 1 : 0);
```

Reply:
done